Change the icon by the achievement status

Laskow opened the pull request:
Change the icon displayed depending on the achievement status, locked or unlocked.

Laskow committed:
Add files via upload

Icons

Laskow committed:
Update ImpAchievements.zs

The new property "sa_Achievement.icon" has been added to all achievements. Also, the description "ia_Melee" was shortened.

Laskow committed:
Update StupidAchievements.zs

Laskow committed:
Added locked/unlocked icon properties

Laskow committed:
Added locked/unlocked icon properties

Laskow committed:
Delete sa_imp100kill.png

Laskow committed:
Delete sa_imp10kill.png

Laskow committed:
Delete sa_imp1kill.png

Laskow committed:
Delete sa_imp666kill.png

Laskow committed:
Delete sa_impfriend.png

Laskow committed:
Delete sa_impmelee.png

Laskow committed:
Delete sa_imptele.png

Laskow committed:
Delete sa_impshadow.png

Laskow committed:
Delete sa_impoverkill.png

Laskow committed:
Add files via upload

Laskow committed:
Update ImpAchievements.zs

mmaulwurff commented on zscript/StupidAchievements.zs:
```diff
@@ -334,6 +341,8 @@ extend class sa_Achievement
   property border        : border;
   property isHidden      : isHidden;
+  property lockedIcon : lockedIcon;
+  property unlockedIcon : unlockedIcon;
```
Align properties, please.

mmaulwurff commented on zscript/StupidAchievements.zs:
```diff
@@ -379,8 +390,11 @@ class sa_ // namespace
+                      , int iconWidth, int iconHeight
                       , double alpha
                       , TextureID texture
+					   , TextureID Icon
```
Here and in other places: please use spaces only, and preserve formatting.

Laskow replied:
These alignments are completely my mistakes. I fixed.

mmaulwurff commented on zscript/StupidAchievements.zs:
```diff
@@ -473,6 +497,7 @@ class sa_Task abstract
   protected int       mNLines;
   protected Font      mFont;
   protected TextureID mTexture;
+  protected TextureID mIcon;
```
mIcon is not initialized, no icon shows on on-screen achievement.

Laskow replied:
Can I ask how to fix it?

mmaulwurff replied:
Assign a value in `sa_Task.init` like you do in `sa_AchievementItem.init`.

Laskow replied:
Thanks! I added some lines. Check the line between 386 – 393 and 497.

mmaulwurff commented on zscript/StupidAchievements.zs:
```diff
@@ -507,15 +532,23 @@ class sa_NoAnimationTask : sa_Task
+	int iconWidth = boxheight;
+	int iconHeight = iconWidth;
+	int iconX =  boxX - iconWidth * 1.1 ;
+	int iconY =  textY - mAchievement.margin / 2;
```
Icon is put to the left of the text box, it won't be displayed if achievements are displayed on the left side of the screen.
In this case, icons should be either displayed to the right of the text box, or text box should be moved to the right.

Laskow replied:
I know that the icon cannot be displayed in notification. Actually I wish to display it like Steam, GOG and other platforms, but I'm worried about it hides status bar and disturbs gameplay.

mmaulwurff replied:
Even without the icon, notification obscures a part of the screen and disturbs gameplay. I think if a player doesn't like that, they can turn of notifications.

> By the way, should the icon be displayed in the notification? Let me know what you think.

So yes, I think it should.

mmaulwurff reviewed (changes requested):
Unfortunately, I cannot accept graphics derived from copyrighted Doom graphics. I suggest making a placeholder icon.

We can include icons in a separate package.

Laskow committed:
Delete sa_Limp100kill.png

Laskow committed:
Delete sa_Limp10kill.png

Laskow committed:
Delete sa_Limp1kill.png

Laskow committed:
Delete sa_Limp666kill.png

Laskow committed:
Delete sa_Limpfriend.png

Laskow committed:
Delete sa_Limpmelee.png

Laskow committed:
Delete sa_Uimp666kill.png

Laskow committed:
Delete sa_Uimp1kill.png

Laskow committed:
Delete sa_Limpoverkill.png

Laskow committed:
Delete sa_Limpshadow.png

Laskow committed:
Delete sa_Limptele.png

Laskow committed:
Delete sa_Uimp100kill.png

Laskow committed:
Delete sa_Uimp10kill.png

Laskow committed:
Delete sa_Uimpfriend.png

Laskow committed:
Delete sa_Uimpmelee.png

Laskow committed:
Delete sa_Uimpoverkill.png

Laskow committed:
Delete sa_Uimpshadow.png

Laskow committed:
Delete sa_Uimptele.png

Laskow committed:
Update StupidAchievements.zs

Laskow committed:
Update StupidAchievements.zs

Laskow committed:
Update StupidAchievements.zs

Laskow committed:
Add files via upload

Laskow committed:
Changed icon file names

Laskow commented:
> Unfortunately, I cannot accept graphics derived from copyrighted Doom graphics. I suggest making a placeholder icon.
> 
> We can include icons in a separate package.

I deleted all the icons that were in conflict with copyright. 
Also I added the new icon named "sa_icon.png" and changed all file names in "ImpAchievements.zs" (I should probably make an icon much better).

Laskow committed:
Fixed file names

Laskow commented:
Sorry, I'm a newbie to GitHub, I didn't know the button "re-request a review" on the right side...

By the way, should the icon be displayed in the notification? Let me know what you think.

Laskow committed:
Now the icon is shown in a notification

But still have a problem with the horizontal position of the icon.

Laskow committed:
Fixed the messe up

Laskow committed:
The icons show both the left side and right sides

Now it's perfect... maybe.

mmaulwurff commented:
Thanks! I'll do some minor adjustments and release a new version of the library.
